mono/utils/dlmalloc.c

```diff
@@ -1312,8 +1312,8 @@ extern void*     sbrk(ptrdiff_t);
 #define MAP_ANONYMOUS        MAP_ANON
 #endif /* MAP_ANON */
 #ifdef MAP_ANONYMOUS
-#define MMAP_FLAGS           (MAP_PRIVATE|MAP_ANONYMOUS)
-#define CALL_MMAP(s)         mmap(0, (s), MMAP_PROT, MMAP_FLAGS, -1, 0)
+#define MONO_MMAP_FLAGS      (MONO_MMAP_PRIVATE|MONO_MMAP_ANON)
+#define CALL_MMAP(s)         mono_valloc (NULL, (s), MONO_MMAP_FLAGS, MONO_MEM_ACCOUNT_DLMALLOC)
```

Comment:
Can you explain why this is necessary?

Reply:
This change was requested by @kumpera to minimize the number of places where we use `mmap` directly (and I suppose also to be able to track the allocs etc)